providers/frame/polkadot_dev.ts

```diff
@@ -16,15 +30,100 @@ export class PolkadotDevProvider extends FrameBinProvider {
     })
   }
 
+  override async chainFile(pathInfo: PathInfo): Promise<File> {
+    const url = new URL(fromPathInfo({ ...pathInfo, filePath: "test-users" }), this.env.href)
+      .toString()
+    const file = await super.chainFile(pathInfo)
+    return new File(`
+      ${file.codeRaw}
+
+      type ArrayOfLength<
+        T,
+        L extends number,
+        A extends T[] = [],
+      > = number extends L ? T[]
+        : L extends A["length"] ? A
+        : ArrayOfLength<T, L, [...A, T]>
+
+      export async function users<N extends number>(count: N): Promise<ArrayOfLength<C.Sr25519, N>>
+      export async function users(count: number): Promise<C.Sr25519[]> {
+        const response = await fetch(
+          "${url}",
+          {
+            method: "POST",
+            headers: {
+              "content-type": "application/json",
+            },
+            body: JSON.stringify({ count }),
+          },
+        )
+        if (response.status >= 400) {
+          throw new Error(await response.text())
+        }
+        const { from, to }: { from: number; to: number } = await response.json()
+        const userIds: C.Sr25519[] = []
+        for (let i = from; i < to; i++) {
+          userIds.push(C.testUser(i))
+        }
+        return userIds
+      }
+    `)
+  }
+
+  override async handle(request: Request, pathInfo: PathInfo): Promise<Response> {
+    if (request.method.toUpperCase() === "POST" && pathInfo.filePath === "test-users") {
+      try {
+        const { count } = await request.json()
+        if (!(count > 0)) {
+          throw new Error("'count' should be greater than 0")
+        }
+        const currentCount = this.#testUserCountCache[this.#getRuntime(pathInfo)] ?? 0
```

Comment:
If `pathInfo.target` is an empty string, that should work fine too Any reason that we'd need to normalize?

Reply:
no special reason, I'll remove the normalization